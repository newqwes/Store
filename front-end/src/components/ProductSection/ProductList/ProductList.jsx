import React from 'react';
import PropTypes from 'prop-types';

import { productsType } from '../../../propType';

import Card from '../../Card';

import ProductListWrapper from './styled';

const ProductList = ({ products, addToCart }) => {
  const goods = products.map((product) => (
    <Card addToCart={addToCart} key={product.id} item={product} />
  ));

  return <ProductListWrapper>{goods}</ProductListWrapper>;
};

ProductList.propTypes = {
  products: productsType.isRequired,
  addToCart: PropTypes.func.isRequired,
};

export default ProductList;
