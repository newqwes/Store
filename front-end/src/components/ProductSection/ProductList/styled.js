import styled from 'styled-components';

const ProductListWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

export default ProductListWrapper;
