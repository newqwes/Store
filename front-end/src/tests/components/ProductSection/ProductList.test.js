import React from 'react';
import { mountWithTheme } from '../../helpers';

import ProductList from '../../../components/ProductSection/ProductList';
import mockProductItem from '../../helpers/mocks';

const setUp = (props) =>
  mountWithTheme(<ProductList products={[mockProductItem]} addToCart={() => {}} {...props} />);

describe('snapshot ProductList', () => {
  const component = setUp();

  it('snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('ProductList component', () => {
  const component = setUp();

  it('should render 1 product', () => {
    const card = component.find('Card');

    expect(card).toHaveLength(1);
  });

  it('should render weight then unitSign', () => {
    const option = component.find('option');
    const text = '250 Ð³';

    expect(option.text()).toBe(text);
  });
});
