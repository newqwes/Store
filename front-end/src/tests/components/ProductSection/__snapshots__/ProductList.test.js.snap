// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot ProductList snapshot 1`] = `
<div
  className="sc-fubCfw fZwJXC"
>
  <div
    className="sc-gKsewC mHKxT"
  >
    <div
      className="sc-bdfBwQ rXYsm"
    >
      <img
        alt="Pizza"
        src="https://image.com"
      />
    </div>
    <div
      className="sc-iBPRYJ jQRuul"
    >
      <span
        className="sc-gsTCUz cakImO"
        fontSize={19}
        fontWeight={500}
      >
        Pizza
      </span>
      <span
        className="sc-gsTCUz bcektt"
        fontSize={12}
        fontWeight={400}
      >
        none description
      </span>
      <div
        className="sc-dlfnbm gnOvZw"
        direction="row"
      >
        <span
          className="sc-jSgupP dMmZuW"
          fontSize={12}
          fontWeight={500}
        >
          24
          руб.
        </span>
        <select
          className="sc-hKgILt dUcerc"
          onChange={[Function]}
        >
          <option
            key="250"
            value={250}
          >
            250
             
            г
          </option>
        </select>
        <button
          className="sc-eCssSg hYdMsO"
          onClick={[Function]}
        >
          В корзину
        </button>
      </div>
    </div>
  </div>
</div>
`;
